client/src/event_loop.rs

```diff
@@ -0,0 +1,269 @@
+use crate::user_input::Inputs;
+use log::debug;
+use std::sync::mpsc::Sender;
+use winit::platform::run_return::EventLoopExtRunReturn;
+use winit::{
+    event::*,
+    event_loop::{ControlFlow, EventLoop},
+    window::{Window, WindowBuilder},
+};
+
+pub struct UserInput {
+    pub(crate) client_id: u32,
```

Comment:
maybe it's not necessary to pack client id to user input. We can attach the client id when sending the command to the server. In the client loop, cliend_id is always fixed, so no need for redundancy.